pkg/transportrequest/uploadcts.go

```diff
@@ -0,0 +1,227 @@
+	action.Application = app
+}
+
+// WithNodeProperties ...
+func (action *CTSUploadAction) WithNodeProperties(node CTSNode) {
+	action.Node = node
+}
+
+// WithTransportRequestID ...
+func (action *CTSUploadAction) WithTransportRequestID(id string) {
+	action.TransportRequestID = id
+}
+
+// WithConfigFile ...
+func (action *CTSUploadAction) WithConfigFile(configFile string) {
+	action.ConfigFile = configFile
+}
+
+// WithDeployUser ...
+func (action *CTSUploadAction) WithDeployUser(deployUser string) {
+	action.DeployUser = deployUser
+}
+
+// Perform ...
+func (action *CTSUploadAction) Perform(command command.ShellRunner) error {
+
+	command.AppendEnv(
+		[]string{
+			fmt.Sprintf("%s=%s", abapUserKey, action.Connection.User),
+			fmt.Sprintf("%s=%s", abapPasswordKey, action.Connection.Password),
+		})
+
+	cmd := []string{"/bin/bash -e"}
+
+	noInstall := len(action.Node.DeployDependencies) == 0
+	if !noInstall {
+		cmd = append(cmd, "echo \"Current user is '$(whoami)'\"")
+		cmd = append(cmd, getPrepareFioriEnvironmentStatement(action.Node.DeployDependencies, action.Node.InstallOpts))
+		cmd = append(cmd, getSwitchUserStatement(action.DeployUser))
+	} else {
+		log.Entry().Info("No deploy dependencies provided. Skipping npm install call. Assuming current docker image already contains the dependencies for performing the deployment.")
+	}
+
+	deployStatement, err := getFioriDeployStatement(action.TransportRequestID, action.ConfigFile, action.Application, action.Connection)
+	if err != nil {
+		return err
+	}
+
+	cmd = append(cmd, deployStatement)
+
+	return command.RunShell("/bin/bash", strings.Join(cmd, "\n"))
+}
+
+func getPrepareFioriEnvironmentStatement(deps []string, npmInstallOpts []string) string {
+	cmd := []string{
+		"npm",
+		"install",
+		"--global",
+	}
+	cmd = append(cmd, npmInstallOpts...)
+	cmd = append(cmd, deps...)
+	return strings.Join(cmd, " ")
+}
+
+func getFioriDeployStatement(
+	transportRequestID string,
+	configFile string,
+	app CTSApplication,
+	cts CTSConnection,
+) (string, error) {
+	desc := app.Desc
+	if len(desc) == 0 {
+		desc = "Deployed with Piper based on SAP Fiori tools"
+	}
+
+	useConfigFile, noConfig, err := handleConfigFile(configFile)
+	if err != nil {
+		return "", err
+	}
+	cmd := []string{
+		"fiori",
+		"deploy",
+		"-f", // failfast --> provide return code != 0 in case of any failure
+		"-y", // autoconfirm --> no need to press 'y' key in order to confirm the params and trigger the deployment
+		"--username", abapUserKey,
+		"--password", abapPasswordKey,
+		"-e", fmt.Sprintf("\"%s\"", desc),
+	}
+
+	if noConfig {
+		cmd = append(cmd, "--noConfig") // no config file, but we will provide our parameters
+	}
+	if useConfigFile {
+		cmd = append(cmd, "-c", fmt.Sprintf("\"%s\"", configFile))
+	}
+	if len(cts.Endpoint) > 0 {
+		log.Entry().Debugf("Endpoint '%s' used from piper config", cts.Endpoint)
+		cmd = append(cmd, "-u", cts.Endpoint)
```

Comment:
Does the endpoint uses the flag `-u`?

Reply:
˙-u˙ is the short flag to add an Endpoint. Took me a minute to understand this. Is there a long form of the flag that is more "aussagekräftig"?

Reply:
Found [the flag descriptions here](https://github.wdf.sap.corp/ux-engineering/tools-suite/tree/master/packages/ui5-tooling#options-1). I would suggest using the long form of the flags:
```suggestion
		cmd = append(cmd, "--url", cts.Endpoint)
```